assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -91,6 +99,101 @@ function MediaProvider({ children }) {
     ]
   );
 
+  const resetMedia = useCallback(
+    ({ pagingNum: p = 1 } = {}) => {
+      fetchMediaStart({ pagingNum: p });
+      getMedia({ mediaType, searchTerm, pagingNum: p })
+        .then(({ data }) => {
+          setMedia({
+            media: data,
+          });
+        })
+        .catch(fetchMediaError);
+    },
+    [
+      setMedia,
+      fetchMediaError,
+      fetchMediaStart,
+      getMedia,
+      mediaType,
+      searchTerm,
+    ]
+  );
+
+  const uploadMediaFromLibrary = useCallback(
+    async (files) => {
+      try {
+        const localMedia = await Promise.all(
+          files.map(getResourceFromLocalFile).reverse()
+        );
+        const filesUploading = files.map((file) => uploadFile(file));
+        setMedia({
+          media: [...localMedia, ...media],
+        });
+        await Promise.all(filesUploading);
```

Comment:
I had an upload fail and it didn't remove the ￼￼￼￼progress bar or remove the element. A finally and or a catch needs to be used here.